kong/plugins/oauth2/access.lua

```diff
@@ -92,10 +92,15 @@ local function get_redirect_uri(client_id)
 end
 
 local function retrieve_parameters()
-  ngx.req.read_body()
-
   -- OAuth2 parameters could be in both the querystring or body
-  return utils.table_merge(ngx.req.get_uri_args(), public_utils.get_body_args())
+  local uri_args = ngx.req.get_uri_args()
+  if ngx.req.get_method() ~= "GET" then
```

Comment:
There are other HTTP verbs that should not carry a body: we should think about `HEAD`, `OPTIONS`, and probably `DELETE` as well.

Reply:
fixed